proto/schema/v1/document.proto

```diff
@@ -0,0 +1,70 @@
+syntax = "proto3";
+
+package sonrio.sonr.schema;
+
+import "schema/v1/schema.proto";
+
+option go_package = "github.com/sonr-io/sonr/x/schema/types";
+option java_multiple_files = true;
+option java_package = "io.sonr.types.schema";
+
+message SchemaDocument {
+  // the DID for this schema should not be populated by request
+  string did = 1;
+
+  string cid = 2;
+
+  string creator = 3;
+
+  // The definition of the document used to build the values
+  SchemaDefinition definition = 4;
+
+  // the properties of this schemadocument
+  repeated SchemaDocumentValue fields = 5;
+}
+
+
+message BoolValue {
+    bool value = 1;
+}
+
+message StringValue {
+    string value = 1;
+}
+
+message IntValue {
+    int32 value = 1;
+}
+
+message FloatValue {
+    double value = 1;
+}
+
+message BytesValue {
+    bytes value = 1;
+}
+
+message LinkValue {
+    LinkKind link = 1;
+    string value = 2;
+}
+
+message ArrayValue {
+    repeated SchemaDocumentValue value = 1;
+}
+
+
+message SchemaDocumentValue {
+    string name = 1;
+    // Type of a single schema property
+    SchemaKind field = 2;
```

Comment:
i think we need an optional field for array item types something like `ItemKind` which would be defined something like the following: 
message ItemKindDefinition {
 SchemaKind type = 1
 ItemKindDefinition item = 2;
}
this definition is recursive to accommodate multidimensional arrays

Reply:
will be addressed in follow up implementation around schemas